src/node_task_runner.cc

```diff
@@ -240,19 +214,60 @@ void ProcessRunner::Run() {
   uv_run(loop_, UV_RUN_DEFAULT);
 }
 
+std::optional<std::tuple<std::filesystem::path, std::string, std::string>>
+FindPackageJson(const std::filesystem::path& cwd) {
+  auto package_json_path = cwd / "package.json";
+  std::string raw_content;
+  std::string path_env_var;
+  auto root_path = cwd.root_path();
+
+  for (auto directory_path = cwd;
+       !std::filesystem::equivalent(root_path, directory_path);
+       directory_path = directory_path.parent_path()) {
+    // Append "path/node_modules/.bin" to the env var, if it is a directory.
+    auto node_modules_bin = directory_path / "node_modules" / ".bin";
+    if (std::filesystem::is_directory(node_modules_bin)) {
+      path_env_var += node_modules_bin.string() + env_var_separator;
+    }
+
+    if (raw_content.empty()) {
+      package_json_path = directory_path / "package.json";
+      // This is required for Windows because std::filesystem::path::c_str()
+      // returns wchar_t* on Windows, and char* on other platforms.
```

Comment:
And it looks to be the correct behaviour as per the standard

https://en.cppreference.com/w/cpp/filesystem/path/native